tests/multidevice/multidevice.cpp

```diff
@@ -224,4 +224,155 @@ TEST_F(multidevice_ft, ensure_color_camera_is_enabled)
     // Make sure we found both devices.
     ASSERT_EQ(master_device_found, true);
     ASSERT_EQ(subordinate_device_found, true);
-}
\ No newline at end of file
+}
+
+typedef struct _parallel_start_data_t
+{
+    k4a_device_t device;
+    k4a_device_configuration_t *config;
+    bool started;
+    LOCK_HANDLE lock;
+} parallel_start_data_t;
+
+static int parallel_start_thread(void *param)
+{
+    parallel_start_data_t *data = (parallel_start_data_t *)param;
+    k4a_result_t result = K4A_RESULT_SUCCEEDED;
+
+    Lock(data->lock);
+    Unlock(data->lock);
+
+    if (!data->started)
+    {
+        EXPECT_EQ(K4A_RESULT_SUCCEEDED, result = k4a_device_start_cameras(data->device, data->config));
+
+        if (K4A_SUCCEEDED(result))
+        {
+            EXPECT_EQ(K4A_RESULT_SUCCEEDED, result = k4a_device_start_imu(data->device));
+        }
+    }
+
+    if (K4A_SUCCEEDED(result))
+    {
+        ThreadAPI_Sleep(1000);
+    }
+
+    if (data->device)
+    {
+        k4a_device_stop_cameras(data->device);
+        k4a_device_stop_imu(data->device);
+        k4a_device_close(data->device);
+        data->device = NULL;
+    }
+
+    return result;
+}
+
+TEST_F(multidevice_ft, start_parallel)
+{
+    LOCK_HANDLE lock;
+    THREAD_HANDLE th1, th2;
+    ASSERT_NE((lock = Lock_Init()), (LOCK_HANDLE)NULL);
+    parallel_start_data_t data1 = { 0 }, data2 = { 0 };
+
+    ASSERT_EQ((uint32_t)2, k4a_device_get_installed_count());
+
+    ASSERT_EQ(K4A_RESULT_SUCCEEDED, k4a_device_open(0, &data1.device));
```

Comment:
Why aren't you doing the open inside the parallel threads?

Reply:
Parallel open is currently not supported. 2 separate LibUsb handles are required and that would need to be serialized. I don't see a great way to implement that yet, other than just creating another global lock.  The solution has to take into account the k4a_device_open, k4a_device_get_installed_count, and AzureKinectFirmwareTool where they each open the two handles in slightly different ways.